src/spikeinterface/core/recording_tools.py

```diff
@@ -306,6 +306,38 @@ def get_chunk_with_margin(
     return traces_chunk, left_margin, right_margin
 
 
+def get_traces_with_margin(
+    recording,
+    segment_index,
+    start_frame,
+    end_frame,
+    channel_indices,
+    margin,
+    add_zeros=False,
+    return_scaled=False,
+    add_reflect_padding=False,
+    window_on_margin=False,
+    dtype=None,
+):
```

Comment:
Maybe a docstring, even if it is similar to get_traces?

Reply:
You are absolutly right, some helper/docs have been added